src/components/Input/Input.vue

```diff
@@ -134,25 +131,25 @@ export default defineComponent({
     outerClassList(): any {
       const baseList = [this.$attrs.class]
       return this.disabled
-        ? ['disabled', ...baseList]
+        ? ['miter-input--disabled', ...baseList]
         : this.invalid
-        ? ['invalid', ...baseList]
-        : this.active
-        ? ['active', ...baseList]
-        : this.hovered
-        ? ['hovered', ...baseList]
-        : baseList
+          ? ['miter-input--invalid', ...baseList]
+          : this.active
+            ? ['miter-input--active', ...baseList]
+            : this.hovered
+              ? ['miter-input--hovered', ...baseList]
+              : baseList
     },
     classList(): any {
       return this.disabled
-        ? ['disabled']
+        ? ['miter-input__input--disabled']
         : this.invalid
-        ? ['invalid']
-        : this.active
-        ? ['active']
-        : this.hovered
-        ? ['hovered']
-        : []
+          ? ['miter-input__input--invalid']
+          : this.active
+            ? ['miter-input__input--active']
+            : this.hovered
+              ? ['miter-input__input--hovered']
+              : []
```

Comment:
Can these be converted to arrays with objects for the dynamic pieces?

Reply:
Yup - updated